Compiler warning in HK unit tests

**Checklist (Please check before submitting)**

* [x] I reviewed the [Contributing Guide](https://github.com/nasa/HK/blob/main/CONTRIBUTING.md).
* [x] I performed a cursory search to see if the bug report is relevant, not redundant, nor in conflict with other tickets.

**Describe the bug**
With sufficient compiler options enabled, I get the following warning:

```
apps/hk/unit-test/hk_app_tests.c:1020:5: error: ‘Buf’ may be used uninitialized [-Werror=maybe-uninitialized]
 1020 |     HK_ResetCountersCmd(&Buf);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~
```

**To Reproduce**
Enable full warnings and at least -O1 optimization level 

**Expected behavior**
Should build cleanly

**System observed on:**
Debian (GCC 12.2)

**Reporter Info**
Joseph Hickey, Vantage Systems, Inc.

--- a/unit-test/hk_app_tests.c
+++ b/unit-test/hk_app_tests.c
@@ -1016,6 +1016,8 @@ void Test_HK_ResetCountersCmd(void)
     char  ExpectedEventString[CFE_MISSION_EVS_MAX_MESSAGE_LENGTH];
     snprintf(ExpectedEventString, CFE_MISSION_EVS_MAX_MESSAGE_LENGTH, "HK Reset Counters command received");
 
+    memset(&Buf, 0, sizeof(Buf));
+
     /* Act */
     HK_ResetCountersCmd(&Buf);
 

